test/compilable/noreturn1.d

```diff
@@ -9,3 +9,13 @@ alias noreturn = typeof(*null);
 pragma(msg, noreturn);
 
 noreturn exits(int* p) { *p = 3; }
+
+noreturn exit();
+
+int test1(int i)
+{
+    if (exit())
+        return i + 1;
+    return i - 1;
+}
```

Comment:
It's actually an optimization. The code gen works without it. It's hard to test optimization patterns. It needs to go in the backend where optimization is done.

`auto return` would type it as `int`. This should be fine.

Reply:
Not parsing unittests is also an optimization, yet not in the backend.
I mentioned `auto return` because having this detected earlier would allow the following:
```D
auto foo ()
{
    if (exit())
       return "I do not matter";
    return 42;
}
```

It's a bit of a dumb example, but at least it's testable.

Reply:
> having this detected earlier would allow the following

That shouldn't be allowed. Optimizations are not supposed to alter the semantics.